In [8]:
import pandas as pd
import statsmodels.formula.api as smf
import statsmodels.api as sm

price = [160,180,200,220,240,260,280]
sale = [126,103,82,75,82,40,20]
cars = [0,9,19,5,25,1,20]
priceDF = pd.DataFrame(price, columns=list('x'))
saleDF = pd.DataFrame(sale, columns=list('y'))
carsDf = pd.DataFrame(cars, columns=list('z'))
houseDf = pd.concat([priceDF,saleDF,carsDf],axis=1)

In [9]:
print(houseDf)

     x    y   z
0  160  126   0
1  180  103   9
2  200   82  19
3  220   75   5
4  240   82  25
5  260   40   1
6  280   20  20


In [16]:
X = houseDf.drop(['y'], axis=1)
y = houseDf.y
Xc = sm.add_constant(X)
linear_regression = sm.OLS(y,Xc)
fitted_model = linear_regression.fit()
fitted_model.summary()

C:\Users\shahn\Anaconda3\lib\site-packages\statsmodels\stats\stattools.py:71: ValueWarning: omni_normtest is not valid with less than 8 observations; 7 samples were given.
  "samples were given." % int(n), ValueWarning)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.919
Model:                            OLS   Adj. R-squared:                  0.879
Method:                 Least Squares   F-statistic:                     22.74
Date:                Fri, 22 Nov 2019   Prob (F-statistic):            0.00654
Time:                        23:53:32   Log-Likelihood:                -25.654
No. Observations:                   7   AIC:                             57.31
Df Residuals:                       4   BIC:                             57.15
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        252.8597     26.812      9.431      0.001     178.417     327.302
x             -0.8249      0.128     -6.445      0.003      -1.180      -0.470
z              0.3593      0.552      0.650      0.551      -1.174       1.893
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   1.646
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.407
Skew:                           0.546   Prob(JB):                        0.816
Kurtosis:                       2.549   Cond. No.                     1.27e+03
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.27e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Coef column gives the value of estimated coefficients (B0, B1, B2 etc.) .

If the coef is zero then that independent variable does not predict the dependent variable correctly.

Std err denotes how much each coefficient varies from the estimated value

t-value - = Estimated coef/stderr

P(>|t|) how likely the estimated value is zero

- This value also indicates how significant a variable is to a model.
- The smaller the value, the more significant a given variable is to the model.
- it is better to remove variables with higher values of `P(>|t|)`

Interpreting the Coef of the Model
The final equation is

y = 126.45 - 0.55 * x - 0.322 * z

the P(>|t|) values for each parameter is

Constant Term 252.85965 is 0.001 - meaning - this term is significant in predicting the output

X - House Price - -0.557 is 0.098 - this term is also significant in predicting the output. z - car sales - 0.322 is 0.668 this term is not so significant in predicting the output.

We do not have to omit the third variable.